Detectors/MUON/MID/Tracking/src/HitMapBuilder.cxx

```diff
@@ -46,61 +67,89 @@ bool HitMapBuilder::crossCommonElement(const std::vector<int>& fired, const std:
   return false;
 }
 
-int HitMapBuilder::getEffFlag(const std::vector<int>& firedRPCLines, const std::vector<int>& nonFiredRPCLines, const std::vector<int>& firedLocIds, const std::vector<int>& nonFiredLocIds) const
+int HitMapBuilder::getEffFlag(const std::vector<int>& firedFEEIdMT11, const std::vector<int>& nonFiredFEEIdMT11) const
 {
+  std::vector<int> firedRPCLines, nonFiredRPCLines;
+  for (auto& feeId : firedFEEIdMT11) {
+    firedRPCLines.emplace_back(detparams::getDEIdFromFEEId(feeId));
+  }
+  for (auto& feeId : nonFiredFEEIdMT11) {
+    nonFiredRPCLines.emplace_back(detparams::getDEIdFromFEEId(feeId));
+  }
+
   if (crossCommonElement(firedRPCLines, nonFiredRPCLines)) {
-    if (crossCommonElement(firedLocIds, nonFiredLocIds)) {
+    if (crossCommonElement(firedFEEIdMT11, nonFiredFEEIdMT11)) {
       return 3;
     }
     return 2;
   }
   return 1;
 }
 
-int HitMapBuilder::getLocId(double xp, double yp, uint8_t deId) const
+int HitMapBuilder::getFEEIdMT11(double xp, double yp, uint8_t deId) const
 {
   auto stripIndex = mMapping.stripByPosition(xp, yp, 0, deId, false);
   if (stripIndex.isValid()) {
-    return mMapping.getBoardId(stripIndex.line, stripIndex.column, deId);
+    auto deIdMT11 = detparams::getDEId(detparams::isRightSide(deId), 0, detparams::getRPCLine(deId));
+    return detparams::getUniqueFEEId(deIdMT11, stripIndex.column, stripIndex.line);
   }
-  return 0;
+  return -1;
+}
+
+bool HitMapBuilder::matchesMaskedChannel(const Cluster& cl) const
+{
+  auto found = mMaskedChannels.find(cl.deId);
+  if (found == mMaskedChannels.end()) {
+    return false;
+  }
+
+  double nSigmas = 4.;
+
+  for (auto& area : found->second) {
+    if (std::abs(cl.xCoor - area.getCenterX()) < nSigmas * cl.getEX() + area.getHalfSizeX() &&
+        std::abs(cl.yCoor - area.getCenterY()) < nSigmas * cl.getEY() + area.getHalfSizeY()) {
+      return true;
+    }
+  }
+  std::cout << cl << std::endl;
+  for (auto& area : found->second) {
+    printf("masked: (%g, %g)  err: (%g, %g)\n", area.getCenterX(), area.getCenterY(), area.getHalfSizeX(), area.getHalfSizeY()); // TODO: REMOVE
+  }
+  return false;
 }
 
 void HitMapBuilder::buildTrackInfo(Track& track, gsl::span<const Cluster> clusters) const
 {
-  std::vector<int> firedLocIds, nonFiredLocIds;
-  std::vector<int> firedDeIds, firedRPCLines, nonFiredRPCLines;
-  bool outsideAcceptance = false;
+  std::vector<int> firedFEEIdMT11, nonFiredFEEIdMT11;
+  bool badForEff = false;
   for (int ich = 0; ich < 4; ++ich) {
     auto icl = track.getClusterMatchedUnchecked(ich);
     if (icl >= 0) {
       auto& cl = clusters[icl];
-      firedDeIds.emplace_back(cl.deId);
-      firedRPCLines.emplace_back(detparams::getRPCLine(cl.deId));
-      firedLocIds.emplace_back(getLocId(cl.xCoor, cl.yCoor, cl.deId));
+      firedFEEIdMT11.emplace_back(getFEEIdMT11(cl.xCoor, cl.yCoor, cl.deId));
       // auto localPt = mHitFinder.getGeometryTransformer().globalToLocal(cl.deId, cl.xCoor, cl.yCoor, cl.zCoor);
```

Comment:
this commented code should be removed, shouldn't it ?

Reply:
Indeed. Will be removed in the next version. I will rewrite the history though so that the two commits will stay separate